internal/infrastructure/ceph/rbd_test.go

```diff
@@ -628,6 +628,141 @@ func TestApplyDiffToRawImage_ChecksumRecoveryAfterCrash(t *testing.T) {
 	assert.Equal(t, []byte("ABCD"), got[:4])
 }
 
+func TestApplyDiffToRawImage_success_RecoverAfterInterruptedCreation(t *testing.T) {
+	// Description:
+	// Verify that applyDiffToRawImage can recreate raw.img when a previous creation attempt
+	// left only the checksum file initialized.
+	//
+	// Arrange:
+	// - raw.img exists but is empty (creation failed before writing data)
+	// - raw.img.csum exists with zero checksums for one expansion unit
+	//
+	// Act:
+	// - Apply a diff that writes to the first chunk
+	//
+	// Assert:
+	// - raw.img is created to the expansion unit size
+	// - the written chunk data and checksum are updated
+	// - untouched chunks keep the zero checksum
+
+	// Arrange
+	rawImageFilePath := getRawImagePathForTest(t)
+	require.NoError(t, os.WriteFile(rawImageFilePath, nil, 0644))
+
+	numChunks := expansionUnitSize / rawChecksumChunkSize
+
+	checksumFilePath := getChecksumFilePath(rawImageFilePath)
+	rawChecksumFile, err := os.OpenFile(checksumFilePath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
+	require.NoError(t, err)
+
+	for i := 0; i < numChunks; i++ {
+		require.NoError(t, writeChecksumAtChunk(rawChecksumFile, uint64(i), csumZero))
+	}
```

Comment:
Let's write the checksum in an incomplete way to test its robustness:


```suggestion
	for i := 0; i < numChunks-1; i++ {
		require.NoError(t, writeChecksumAtChunk(rawChecksumFile, uint64(i), csumZero))
	}
```